compiler/codegen/OMRCodeGenerator.cpp

```diff
@@ -3017,7 +3017,7 @@ OMR::CodeGenerator::addStaticRelocation(const TR::StaticRelocation &relocation)
 intptrj_t OMR::CodeGenerator::hiValue(intptrj_t address)
    {
    if (self()->comp()->compileRelocatableCode()) // We don't want to store values using HI_VALUE at compile time, otherwise, we do this a 2nd time when we relocate (and new value is based on old one)
-      return ((address >> 16) & 0x0000FFFF);
+      return address >> 16;
```

Comment:
This is an intentional change to the behaviour of this function to ensure that the sign bit is not cleared. AFAICT the result of this function is always used as the immediate of an `addis`/`lis` instruction or in a context where the upper bits are ignored. Since `addis`/`lis` use a signed immediate and the new binary encoder will validate that these immediates are within the expected range, masking off the sign bit of a negative value will produce an out-of-range immediate and trigger a crash.

This change shouldn't break anything that wasn't already broken when not running under AOT, since in that case the `HI_VALUE` macro will be used instead, which also does not mask its result.